### PR TITLE
Application Mobility Adjacent and Mobility procedure sub/notifications

### PR DESCRIPTION
**Pull request by pastorsx**
Implementation of all endpoints except GET adjacent App nodes

testing:
UT passed
system passed

### CODE REVIEW

**Review by dilallkx — APPROVED** [review]

File: go-apps/meep-ams/server/ams.go [hunk 1]
```diff
@@ -1678,3 +1949,82 @@ func updateStoreName(storeName string) {
...
+			//check apps first
+			key := baseKeyNoMep + "*:apps:*:dev:" + address
+			err := rc.ForEachEntry(key, populateAppInstanceIds, &appInstanceIdsList)
+			if err != nil {
+				log.Error(err)
+				return
+			}
+			//if no single app, seach for whole mep
+			if len(appInstanceIdsList.AppInstanceIds) == 0 {
+				key = baseKeyNoMep + "*:mep:*:dev:" + address
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-ams/server/ams.go:1983]: using baseKeyGlobal

File: go-apps/meep-ams/server/ams.go [hunk 2]
```diff
@@ -774,18 +971,33 @@ func repopulateMpSubscriptionMap(key string, jsonInfo string, userData interface
...
+	}
+	return false
+}
+
 func isMatchFilterCriteriaAppInsId(subscriptionType string, filterCriteria interface{}, appId string) bool {
 	switch subscriptionType {
 	case MOBILITY_PROCEDURE_SUBSCRIPTION:
 		return isMatchMpFilterCriteriaAppInsId(filterCriteria, appId)
+	case ADJACENT_APP_INFO_SUBSCRIPTION:
+		return isMatchAdjFilterCriteriaAppInsId(filterCriteria, appId)
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-ams/server/ams.go:997]: could have... left it as is

File: go-apps/meep-ams/server/ams.go [hunk 3]
```diff
@@ -665,6 +709,159 @@ func mec011AppTerminationPost(w http.ResponseWriter, r *http.Request) {
...
+	if string(*appInfo1.State) != string(*appInfo2.State) {
+		return true
+	}
+	return false
+}
+
+func checkAdjAppInfoNotificationRegisteredSubscriptions(appNames []string) {
+
+	mutex.Lock()
+	defer mutex.Unlock()
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-ams/server/ams.go:746]: removed

File: go-apps/meep-ams/server/ams.go [hunk 4]
```diff
@@ -237,6 +266,26 @@ func Init() (err error) {
...
+		}
+	}
+	log.Info("PERIODIC_TRIGGER_INTERVAL: ", periodicTriggerInterval)
+
+	periodicTriggerTicker = time.NewTicker(time.Duration(periodicTriggerInterval) * time.Second)
+	go func() {
+		for range periodicTriggerTicker.C {
+			checkPeriodicTrigger()
+		}
+	}()
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-ams/server/ams.go:287]: moved

File: go-apps/meep-ams/server/ams.go [hunk 5]
```diff
@@ -122,11 +131,16 @@ var registrationTicker *time.Ticker
 var amsMqLocal *mq.MsgQueue
 var amsApiMgr *sam.SwaggerApiMgr
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-ams/server/ams.go:132]: done

File: go-apps/meep-ams/server/ams.go [hunk 6]
```diff
@@ -218,6 +246,7 @@ func Init() (err error) {
 
 	// Set base storage key
 	baseKey = dkm.GetKeyRoot(sandboxName) + amsKey + ":mep:" + mepName + ":"
+	baseKeyNoMep = dkm.GetKeyRoot(sandboxName) + amsKey + ":"
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-ams/server/ams.go:249]: done

File: go-apps/meep-ams/sbi/ams-sbi.go [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+				found = true
+				break
+			}
+		}
+		if !found {
+			sbi.updateDeviceInfoCB(prevUeName, "", nil)
+		}
+	}
+
+	// find all apps under our mep
```
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-ams/sbi/ams-sbi.go:263]: not sure i get the question...

[the remark this answers is not in the document]
Comment (dilallkx) [reply go-apps/meep-ams/sbi/ams-sbi.go:263]: There is no code after the comment. Should the comment be removed?

File: go-apps/meep-ams/server/ams.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (dilallkx) [reply go-apps/meep-ams/server/ams.go:962]: ok

File: go-apps/meep-ams/sbi/ams-sbi.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (pastorsx) [reply go-apps/meep-ams/sbi/ams-sbi.go]: comment removed

**Pull request merged**